networking/configuring-ipfailover.adoc

```diff
@@ -0,0 +1,62 @@
+[id="configuring-ipfailover"]
+= Configuring IP failover
+include::modules/common-attributes.adoc[]
+:context: configuring-ipfailover
+
+toc::[]
+
+This topic describes configuring IP failover for pods and services on your {product-title} cluster.
+
+IP failover manages a pool of Virtual IP (VIP) addresses on a set of nodes. Every VIP in the set is serviced by a node selected from the set. As long a single node is available, the VIPs is served. There is no way to explicitly distribute the VIPs over the nodes. so there can be nodes with no VIPs and other nodes with many VIPs. If there is only one node, all VIPs are on it.
+
+[NOTE]
+====
+The VIPs must be routable from outside the cluster.
+====
+
+IP failover monitors a port on each VIP to determine whether the port is reachable on the node. If the port is not reachable, the VIP is not assigned to the node. If the port is set to `0`, this check is suppressed. The check script does the needed testing.
+
+IP failover uses link:http://www.keepalived.org/[Keepalived] to host a set of externally accessible VIP addresses on a set of hosts. Each VIP is only serviced by a single host at a time. Keepalived uses the Virtual Router Redundancy Protocol (VRRP) protocol to determine which host, from the set of hosts, services which VIP. If a host becomes unavailable or if the service that Keepalived is watching does not respond, the VIP is switched to another host from the set, which means a VIP is always serviced as long as a host is available.
+
+When a host running Keepalived passes the check script, the host can become in the master state based on its priority and the priority of the current master, as determined by the preemption strategy.
+
+A cluster administrator can provide a script though the `OPENSHIFT_HA_NOTIFY_SCRIPT` variable, which is called whenever the state changes. Keepalived is in master state when it is servicing the VIP, in backup state when another node is servicing the VIP, or in fault state when the check script fails. The notify script is called with the new state whenever the state changes.
```

Comment:
"though" → "through".

"which" is ambiguous.

Need to make sure we have clear and consistent terminology and notation.
```suggestion
A cluster administrator can provide a script through the `OPENSHIFT_HA_NOTIFY_SCRIPT` variable, and this script is called whenever the state of the VIP on the node changes. Keepalived uses the `master` state when it is servicing the VIP, the `backup` state when another node is servicing the VIP, or in the `fault` state when the check script fails. The notify script is called with the new state whenever the state changes.
```